### PR TITLE
Apply stricter lint rules

### PR DESCRIPTION
**Pull request by ghost**
This is in preparation of tightening up the linting. It applies more of tslint's core rules, and also lints that user URLs should be for `https://github.com` and not for `http://github.com` or a personal website.
I have tested this locally; travis will likely time out.

### CODE REVIEW

**Review by thSoft — APPROVED** [review]
Modification of types/angularfire/index.d.ts is OK.

**Review by jpmnteiro — APPROVED** [review]

**Review by jpmnteiro — APPROVED** [review]
Modification of types/angular-gridster/index.d.ts is OK.

**Review by bastienmoulia — APPROVED** [review]
Modification of types/angular-pdfjs-viewer/index.d.ts is OK.

**Review by aciccarello — APPROVED** [review]
:+1: on angular-cookies changes

File: types/angular-block-ui/index.d.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ declare module 'angular' {
              * @param {angular.IRequestConfig} config - the Angular request config object.
```
Comment (sclassen) [inline types/angular-block-ui/index.d.ts]: should the `angular.` also be removed from the TS Doc?

Comment (andy-ms) [inline types/angular-block-ui/index.d.ts]: Some day I'd like to go through everything and remove redundant JSDoc type annotations.

**Comment by sclassen** [comment]
:+1: on block-ui changes

**Comment by alvis** [comment]
The removal of `no-var` rule in _node/tslint.json_ has caused a cascaded failing on type checking. While a supposed-to-be-working `no-var-keyword` tslint rule is added to the file amid the removal, the test doesn't honour it.

**Comment by andy-ms** [comment]
@alvis Oh, that's because I need to upload the updated dtslint. `no-var` was a workaround for a bug in `no-var-keyword` that's been fixed in the latest tslint. Thanks!

**Comment by kjin** [comment]
@andy-ms What's the status on the dtslint update? It seems to be blocking #18952 and other PRs against @types/node.

**Comment by andy-ms** [comment]
@kjin Will do that today!

**Comment by alvis** [comment]
Thanks @andy-ms! 👍 
Once it is fixed, #19114 can move forward too.

**Comment by andy-ms** [comment]
dtslint updated.

**Comment by kjin** [comment]
@andy-ms For #18952 I think CI is failing due to unrelated failures ([link](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/266982102)), do you know why this might be the case?

**Pull request merged**